include/storage/io.hpp

```diff
@@ -52,23 +179,20 @@ static_assert(sizeof(HSGRHeader) == 20, "HSGRHeader is not packed");
 
 // Reads the checksum, number of nodes and number of edges written in the header file of a `.hsgr`
 // file and returns them in a HSGRHeader struct
-inline HSGRHeader readHSGRHeader(boost::filesystem::ifstream &input_stream)
+inline HSGRHeader readHSGRHeader(io::File &input_file)
 {
     const util::FingerPrint fingerprint_valid = util::FingerPrint::GetValid();
-    util::FingerPrint fingerprint_loaded;
-    input_stream.read(reinterpret_cast<char *>(&fingerprint_loaded), sizeof(util::FingerPrint));
+    const auto fingerprint_loaded = input_file.readOne<util::FingerPrint>();
     if (!fingerprint_loaded.TestGraphUtil(fingerprint_valid))
     {
         util::SimpleLogger().Write(logWARNING) << ".hsgr was prepared with different build.\n"
                                                   "Reprocess to get rid of this warning.";
     }
 
     HSGRHeader header;
-    input_stream.read(reinterpret_cast<char *>(&header.checksum), sizeof(header.checksum));
-    input_stream.read(reinterpret_cast<char *>(&header.number_of_nodes),
-                      sizeof(header.number_of_nodes));
-    input_stream.read(reinterpret_cast<char *>(&header.number_of_edges),
-                      sizeof(header.number_of_edges));
+    input_file.readInto(header.checksum);
+    input_file.readInto(header.number_of_nodes);
+    input_file.readInto(header.number_of_edges);
```

Comment:
We had a policy-based writer once abstracting over all of this (never got merged..)
https://github.com/daniel-j-h/osrm-backend/blob/39383855dd6958a0de3b9981b2780b4a37ce4b56/io/osrm_writer.hpp

Why do we have to know about the file's layout here?